src/xtgeo/grid3d/_grid_hybrid.py

```diff
@@ -53,6 +52,10 @@ def make_hybridgrid(
 
     del rvalues
 
+    # when a hybridgrid is made, the current subrid settings lose relevance, hence
+    # it is forced set to None
+    self.subgrids = None
```

Comment:
Have to say, it wasn't obvious what a hybridgrid is. The documentation found in python didn't provide that much help, and the c documentation was a bit vague
>Convert a 3D grid to a hybrid grid; i.e. horisontal layers between two levels

Considering that there are two c functions made for this, and each have about 200-300 lines of code each I would argue that we should have some test coverage on this. Given the two test functions (`test_hybridgrid1` and `test_hybridgrid2`) made, we might need to expand those a bit.

This would be an excellent opportunity to do so in my opinion!

Reply:
It is ~documented in this example: https://xtgeo.readthedocs.io/en/stable/usextgeoroxar.html#make-a-hybrid-grid

I can see what I can do 👍 in order to improve the docs / tests a bit.

Reply:
Ah, I didn't even look in the examples.. Thanks

> I can see what I can do 👍 in order to improve the docs / tests a bit.

Perfect :)

Reply:
Just a quick peak of the documentation (the image definitively helped!), and I think this deserves to be well tested. I can help out though, writing tests..

Could I ask, what are the use cases? Is it much used? Can it be used as a sort of LGR, only with horizontal lines?

Reply:
The major use case (to my knowledge) is the Troll model where it is used actively (but I think there may be other fields also?). See this paper which describes it well: https://onepetro.org/SPERCSC/proceedings/11RCSC/All-11RCSC/SPE-148023-MS/150366

It replaces numerous (and conflicting) LGR's. The method could of course exist in RMS, but since we found a solution that works we haven't pushed it... 

> How are cells split up at the boundaries?

Not easy to explain by text; I need to "draw and tell"...

Reply:
Yes more tests would be beneficial. I.e. the subgrid issue was not catched by tests, and Troll found it the "hard way". An issue (which is not in the issue list) is to be able to run this a multiple time in order to make regional hybrid grids at different vertical levels and resolutions. Currently this will multiply the number of vertical cells which not optimal.

Reply:
Thanks for the paper, that provided some insight on the use case.

> The method could of course exist in RMS, but since we found a solution that works we haven't pushed it...

The reason I asked was because the example was in "usextgeoroxar", but good to know. Sounds like this is a fairly field specific feature.

Could make some simple tests where the original grid is already rectangular (should work then as well I suppose), and verify behaviour when hybridgrid is made at the boundary and when it's in the middle of a cell.